src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1107 @@
+         end if
+
+         deallocate(fr)
+
+      else
+
+         ! calculate the rate constant scalar for soil water content.
+         ! Uses the log relationship with water potential given in
+         ! Andren, O., and K. Paustian, 1987. Barley straw decomposition in the field:
+         ! a comparison of models. Ecology, 68(5):1190-1200.
+         ! and supported by data in
+         ! Orchard, V.A., and F.J. Cook, 1983. Relationship between soil respiration
+         ! and soil moisture. Soil Biol. Biochem., 15(4):447-453.
+
+         do j = 1,nlevdecomp
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               psi = min(soilpsi(c,j),maxpsi)
+               ! decomp only if soilpsi is higher than minpsi
+               if (psi > minpsi) then
+                  w_scalar(c,j) = (log(minpsi/psi)/log(minpsi/maxpsi))
+               else
+                  w_scalar(c,j) = 0._r8
+               end if
+            end do
+         end do
+
+         if (use_lch4) then
+            ! Calculate ANOXIA
+            ! Check for anoxia w/o LCH4 now done in controlMod.
+
+            if (anoxia) then
+               do j = 1,nlevdecomp
+                  do fc = 1,num_soilc
+                     c = filter_soilc(fc)
+
+                     o_scalar(c,j) = max(o2stress_unsat(c,j), mino2lim)
+                  end do
+               end do
+            else
+               o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+            end if
+         else
+            o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+         end if
+
+      end if
+
+      ! Term that reduces decomposition rate at depth
+      do j = 1, nlevdecomp
+         do fc = 1, num_soilc
+            c = filter_soilc(fc)
+            if (use_vertsoilc) then
+               ! Using fixed e-folding depth as in
+               ! SoilBiogeochemDecompCascadeBGCMod.F90
+               depth_scalar(c,j) = exp(-zsoi(j) / decomp_depth_efolding)
+            else
+               depth_scalar(c,j) = 1.0_r8
+            end if
+         end do
+      end do
+
+      ! TODO Move to the params files
+      fmet_p1 = 0.85_r8
+      fmet_p2 = 0.85_r8
+      fmet_p3 = 0.013_r8
+      fmet_p4 = 40.0_r8
+      fchem_r_p1 = 0.1_r8
+      fchem_r_p2 = -3.0_r8
+      fchem_k_p1 = 0.3_r8
+      fchem_k_p2 = -3.0_r8
+      tauMod_factor = 0.01_r8  ! was tauMod_denom = 100
+      tauMod_min = 0.8_r8
+      tauMod_max = 1.2_r8
+      tau_r_p1 = 5.2e-4_r8
+      tau_r_p2 = 0.3_r8  ! or 0.4?
+      tau_k_p1 = 2.4e-4_r8
+      tau_k_p2 = 0.1_r8
+      ko_r = 4.0_r8
+      ko_k = 4.0_r8
+      densdep = 1.0_r8
+      desorpQ10 = 1.1_r8
+      t_soi_ref = 25.0_r8  ! deg C
+
+      ! calculate rates for all litter and som pools
+      ! TODO Ok that I reversed order of do-loops?
+      do fc = 1,num_soilc
+         c = filter_soilc(fc)
+
+         ! Time-dependent params from Wieder et al. 2015 & testbed code
+         ! TODO STILL MISSING N-related stuff: DIN...
```

Comment:
start looking in NitrogenCompetitionMod:
we will need hooks to the ctsm no3 and nh4 terms
the fluxes are nmin (nh4)